src/number_theory/modular_forms/jacobi_theta.lean

```diff
@@ -92,3 +110,127 @@ begin
+
+/-- An explicit upper bound for `‖jacobi_theta τ - 1‖`. -/
+lemma jacobi_theta_sub_one_norm_le (τ : ℍ) :
+  ‖jacobi_theta τ - 1‖ ≤ 2 / (1 - exp (-π * τ.im)) * exp (-π * τ.im) :=
+begin
+  suffices : ‖∑' (n : ℕ), cexp (π * I * (n + 1) ^ 2 * τ)‖ ≤ exp (-π * τ.im) / (1 - exp (-π * τ.im)),
+  { calc ‖jacobi_theta τ - 1‖ = 2 * ‖∑' (n : ℕ), cexp (π * I * (n + 1) ^ 2 * τ)‖ :
+      by rw [sub_eq_iff_eq_add'.mpr (jacobi_theta_eq_tsum_nat τ), norm_mul, complex.norm_eq_abs,
+        complex.abs_two]
+    ... ≤ 2 * (rexp (-π * τ.im) / (1 - rexp (-π * τ.im))) :
+      by rwa [mul_le_mul_left (zero_lt_two' ℝ)]
+    ... = 2 / (1 - rexp (-π * τ.im)) * rexp (-π * τ.im) : by rw [div_mul_comm, mul_comm] },
+  have : ∀ (n : ℕ), ‖cexp (π * I * (n + 1) ^ 2 * τ)‖ ≤ exp (-π * τ.im) ^ (n + 1),
+  { intro n,
+    rw [(by { push_cast, ring } : ↑π * I * (n + 1) ^ 2 * τ = ↑(π * (n + 1) ^ 2 : ℝ) * (τ * I)),
+      complex.norm_eq_abs, complex.abs_exp, of_real_mul_re, mul_I_re, upper_half_plane.coe_im,
+      mul_neg, ←neg_mul, ←neg_mul, mul_assoc, mul_comm, mul_assoc, mul_comm τ.im,
+      (by push_cast : ((n:ℝ) + 1) ^ 2 = ↑((n + 1) ^ 2)), real.exp_nat_mul],
+    exact pow_le_pow_of_le_one (exp_pos _).le
+      (exp_le_one_iff.mpr $ (mul_neg_of_neg_of_pos (neg_lt_zero.mpr pi_pos) τ.im_pos).le)
+      (nat.le_self_pow two_ne_zero _) },
+  have s : has_sum (λ n : ℕ, rexp (-π * τ.im) ^ (n + 1))
+    (rexp (-π * τ.im) / (1 - rexp (-π * τ.im))),
+  { simp_rw [pow_succ, div_eq_mul_inv, has_sum_mul_left_iff (real.exp_ne_zero _)],
+    exact has_sum_geometric_of_lt_1 (exp_pos (-π * τ.im)).le
+      (exp_lt_one_iff.mpr $ (mul_neg_of_neg_of_pos (neg_lt_zero.mpr pi_pos) τ.im_pos)) },
+  have aux : summable (λ (n : ℕ), ‖cexp (↑π * I * (↑n + 1) ^ 2 * ↑τ)‖),
+    from summable_of_nonneg_of_le (λ n, norm_nonneg _) this s.summable,
+  exact (norm_tsum_le_tsum_norm aux).trans
+    ((tsum_mono aux s.summable this).trans (le_of_eq s.tsum_eq)),
+end
+
+/-- The norm of `jacobi_theta τ - 1` decays exponentially as `im τ → ∞`. -/
+lemma jacobi_theta_sub_one_is_O_im_infty :
+  is_O upper_half_plane.at_im_infty (λ τ, jacobi_theta τ - 1) (λ τ, rexp (-π * τ.im)) :=
+begin
+  simp_rw [is_O, is_O_with, filter.eventually, upper_half_plane.at_im_infty_mem],
+  refine ⟨2 / (1 - rexp (-π)), 1, (λ τ hτ, (jacobi_theta_sub_one_norm_le τ).trans _)⟩,
+  rw [real.norm_eq_abs, real.abs_exp],
+  refine mul_le_mul_of_nonneg_right _ (exp_pos _).le,
+  rw [div_le_div_left (zero_lt_two' ℝ), sub_le_sub_iff_left, exp_le_exp, neg_mul, neg_le_neg_iff],
+  { exact le_mul_of_one_le_right pi_pos.le hτ },
+  { rw [sub_pos, exp_lt_one_iff, neg_mul, neg_lt_zero], exact mul_pos pi_pos τ.im_pos },
+  { rw [sub_pos, exp_lt_one_iff, neg_lt_zero], exact pi_pos }
+end
+
+-- Formulation of this result is somewhat roundabout, since functions on subtypes don't play well
+-- with `differentiable_at`.
+lemma jacobi_theta_differentiable_at (τ : ℍ) :
+  differentiable_at ℂ (λ z, ∑' (n : ℤ), cexp (π * I * n ^ 2 * z)) ↑τ :=
```

Comment:
It might be worth adding a version of this for `mdifferentiable` (as functions `ℍ → ℂ`) since this is what the modular forms class will want.

Reply:
OK, I have pushed a new version. I added a new general lemma saying that open embeddings are `mdifferentiable`, and derived it from that.